src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkUnusedLogSectionInspection.kt

```diff
@@ -0,0 +1,72 @@
+package com.jetbrains.snakecharm.inspections
+
+import com.intellij.codeInspection.*
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Key
+import com.intellij.psi.util.parentOfType
+import com.jetbrains.snakecharm.SnakemakeBundle
+import com.jetbrains.snakecharm.lang.SnakemakeNames
+import com.jetbrains.snakecharm.lang.psi.*
+
+class SmkUnusedLogSectionInspection : SnakemakeInspection() {
+    companion object {
+        val KEY = Key<HashSet<SmkRuleOrCheckpoint>>("SmkLogUnusedSection_Rules")
+    }
+
+    override fun buildVisitor(
+        holder: ProblemsHolder,
+        isOnTheFly: Boolean,
+        session: LocalInspectionToolSession
+    ) = object : SnakemakeInspectionVisitor(holder, session) {
+
+        override fun visitSmkRule(rule: SmkRule) {
+            visitSMKRuleLike(rule)
+        }
+
+        override fun visitSmkCheckPoint(checkPoint: SmkCheckPoint) {
+            visitSMKRuleLike(checkPoint)
+        }
+
+        private fun visitSMKRuleLike(rule: SmkRuleLike<SmkArgsSection>) {
+            val logSection = rule.getSections().firstOrNull {
+                it.sectionKeyword == SnakemakeNames.SECTION_LOG
+            } ?: return
+            val sectionsByRule = session.putUserDataIfAbsent(KEY, hashSetOf())
+
+            if (rule in sectionsByRule) {
+                return
```

Comment:
Not sure that is actually needed. What is used for? Is it caching that allows next time not to check the rule? What if rule has changed? Inspections will be re-run automatically, but cache could be out-of-data. Not 100% sure, just my guess. Maybe I didn't get the idea.

Reply:
As I understand, `sectionsByRule` will be overwritten before each inspection call (in debug mode, `session` saves rules we need while it checks file rules and then, befor next file inspection, it forgots all rules). During one inspection session code can't be modified, so at this point everything should be ok. 
Every time inspection visits rule statement, it scans all available rules, so we use such check in order not to call scanning (via `SmkSLSectionReferencesCollector`) for rules about which we know, that they were referenced during current inspection session